portals/knowledge-management/src/Pages/ViewCategory.vue

```diff
@@ -193,22 +210,31 @@
             debounceSearch(searchQuery.value, page);
             return;
         }
+
         loadingResults.value = true;
 
         const { get } = consumer();
+        const slug = route.params.categorySlug ?? route.params.subCategorySlug;
 
-        await get(props.apiUrl + '/categories/' + route.params.categorySlug, { page: page, filter: filter.value }).then(
-            (response) => {
-                if (response.data.category.slug !== route.params.categorySlug) {
+        await get(props.apiUrl + '/categories/' + slug, { page: page, filter: filter.value }).then((response) => {
+            if (response.data.category.slug !== slug) {
+                if (route.params.categorySlug) {
                     router.replace({ name: 'view-category', params: { categorySlug: response.data.category.slug } });
+                } else {
+                    router.replace({
+                        name: 'view-subcategory',
+                        params: { subCategorySlug: response.data.category.slug },
+                    });
                 }
+            }
 
-                category.value = response.data.category;
-                articles.value = response.data.articles.data;
-                setPagination(response.data.articles);
-                loadingResults.value = false;
-            },
-        );
+            category.value = response.data.category;
+            subCategories.value = response.data.subCategories;
+            parentCategory.value = response.data.parentCategory;
```

Comment:
Rather than having all these seperate values for `subCategories` and `parentCategory` I think it may be easier to parse if that data is just stored as optional/nullable values in the `category`. And then just use it / modify the experience if they are present.